test/grant-api.test.js

```diff
@@ -0,0 +1,105 @@
+import { describe, it, before, after, beforeEach } from "node:test";
+import { assert } from "../src/common/assert.js";
+import { DockerComposeEnvironment, Wait } from "testcontainers";
+import Wreck from "@hapi/wreck";
+import { MongoClient } from "mongodb";
+import { config } from "../src/common/config.js";
+import { grant1, grant2 } from "./fixtures/grants.js";
+import {
+  toGrant,
+  collection as grantsCollection,
+} from "../src/grants/grant-repository.js";
+
+describe("Grant API Tests", { timeout: 120000 }, function () {
+  const PORT = 3001;
+  const MONGO_PORT = 28017;
+  let appUrl;
+  let environment;
+  let mongoUri;
+  let db;
+  let client;
+
+  before(async () => {
+    environment = await new DockerComposeEnvironment(".", "compose.yml")
+      .withEnvironment({ PORT: PORT.toString() })
+      .withEnvironment({ MONGO_PORT: MONGO_PORT.toString() })
+      .withWaitStrategy("mongodb", Wait.forHealthCheck())
+      .withWaitStrategy("gas", Wait.forLogMessage("server started"))
+      .up();
+    console.log("Containers are up and running.");
+
+    // Get MongoDB container URI
+    const container = environment.getContainer("mongodb-1");
+    const mongoPort = container.getMappedPort(MONGO_PORT);
+    const host = container.getHost();
+    mongoUri = `mongodb://${host}:${mongoPort}`;
+    appUrl = `http://${host}:${PORT}`;
+
+    // Connect to MongoDB
+    client = new MongoClient(mongoUri);
+    await client.connect();
+    db = client.db(config.get("mongoDatabase"));
+    console.log("Connected to MongoDB.");
+  });
+
+  after(async () => {
+    // Shutdown and cleanup the environment
+    if (client) await client.close();
+    if (environment) await environment.stop();
+    console.log("Containers stopped and cleaned up.");
+  });
+
+  describe("POST /grants", () => {
+    beforeEach(async () => {
+      await db.collection(grantsCollection).deleteMany({});
+    });
+
+    it("Add a grant", async () => {
```

Comment:
To stay consistent with the rest of the codebase:
```suggestion
    it("adds a grant", async () => {
```